npm/private/npm_package_store.bzl

```diff
@@ -195,17 +195,41 @@ def _npm_package_store_impl(ctx):
             virtual_store_directory = src_directory
         else:
             virtual_store_directory = ctx.actions.declare_directory(virtual_store_directory_path)
-            copy_directory_bin_action(
-                ctx,
-                src = src_directory,
-                dst = virtual_store_directory,
-                copy_directory_bin = ctx.toolchains["@aspect_bazel_lib//lib:copy_directory_toolchain_type"].copy_directory_info.bin,
-                # Hardlinking source files in external repositories as was done under the hood
-                # prior to https://github.com/aspect-build/rules_js/pull/1533 may lead to flaky build
-                # failures as reported in https://github.com/aspect-build/rules_js/issues/1412.
-                hardlink = ctx.attr.hardlink,
-                verbose = ctx.attr.verbose,
-            )
+            if src_directory.path.endswith(".tgz"):
```

Comment:
is it always `.tgz`? can it end with `.tar.gz`?

Reply:
We control the name of the archive when we download it so it will always be `.tgz` but its a good call to be more robust... better test would be to assume its an archive to extract if the final segment of the path does not end with `package` which is the name of the directory we extract to in npm_import.

Reply:
can we just do check if the src_directory (naming?) is a directory by doing `src_directory.is_directory`?

Reply:
It doesn't work since `is_directory` only works for output files. Bazel doesn't track if a source file is a directory or a file. Opted to add a is_tarball util function to check against all of the possible common extensions.